src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -0,0 +1,208 @@
+package frc.robot.subsystems;
+
+import com.chopshop166.chopshoplib.commands.CommandChain;
+import com.chopshop166.chopshoplib.outputs.SendableSpeedController;
+import com.chopshop166.chopshoplib.sensors.Lidar;
+
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.GenericHID.Hand;
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.InstantCommand;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.interfaces.Gyro;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.Robot;
+import frc.robot.RobotMap;
+
+public class Drive extends Subsystem {
+
+    private SendableSpeedController left;
+    private SendableSpeedController right;
+    private DoubleSolenoid climbPiston;
+    private Lidar lidar;
+    private Encoder leftEncoder;
+    private Encoder rightEncoder;
+    private Gyro gyro;
+    private DifferentialDrive drive = new DifferentialDrive(left, right);
+
+    public Drive(final RobotMap.DriveMap map) { // NOPMD
+        super();
+        // Take values that the subsystem needs from the map, and store them in the
+        // class
+
+        left = map.getLeft();
+        right = map.getRight();
+        climbPiston = map.getClimbPiston();
+        lidar = map.getLidar();
+        leftEncoder = map.getLeftEncoder();
+        rightEncoder = map.getRightEncoder();
+        gyro = map.getGyro();
+    }
+    // TODO put numbers here
+
+    double gyroCorrection;
+
+    PIDSource gyroSource = new PIDSource() {
+
+        @Override
+        public void setPIDSourceType(PIDSourceType pidSource) {
+
+        }
+
+        @Override
+        public double pidGet() {
+            return gyro.getAngle();
+        }
+
+        @Override
+        public PIDSourceType getPIDSourceType() {
+            return null;
+        }
+    };
+
+    PIDController gyroDrivePID = new PIDController(.01, .0009, 0.0, 0.0, gyroSource, (double value) -> {
+        gyroCorrection = value;
+    });
+
+    double sandstormSpeed = .2;
+
+    @Override
+    public void initDefaultCommand() {
+
+        // Set the default command for a subsystem here.
+        // setDefaultCommand(new MySpecialCommand());
```

Comment:
You should set `driveNormal` as the default command.

Reply:
·